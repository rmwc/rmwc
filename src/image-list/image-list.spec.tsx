import React from 'react';
import { render } from '@testing-library/react';
import {
  ImageList,
  ImageListItem,
  ImageListImageAspectContainer,
  ImageListImage,
  ImageListSupporting,
  ImageListLabel
} from './';

const images = [
  'https://material-components-web.appspot.com/images/photos/3x2/1.jpg',
  'https://material-components-web.appspot.com/images/photos/2x3/1.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/4.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/5.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/6.jpg',
  'https://material-components-web.appspot.com/images/photos/2x3/2.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/8.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/11.jpg',
  'https://material-components-web.appspot.com/images/photos/2x3/5.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/13.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/14.jpg',
  'https://material-components-web.appspot.com/images/photos/2x3/6.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/15.jpg',
  'https://material-components-web.appspot.com/images/photos/3x2/16.jpg',
  'https://material-components-web.appspot.com/images/photos/2x3/7.jpg'
];

describe('ImageList', () => {
  test('renders', () => {
    const { asFragment } = render(
      <ImageList>
        {images.map((src) => (
          <ImageListItem
            key={src}
            style={{ margin: '2px', width: 'calc(100% / 5 - 4.2px)' }}
          >
            <ImageListImageAspectContainer
              style={{ paddingBottom: 'calc(100% / 1.5)' }}
            >
              <ImageListImage src={src} />
            </ImageListImageAspectContainer>
            <ImageListSupporting>
              <ImageListLabel>Text label</ImageListLabel>
            </ImageListSupporting>
          </ImageListItem>
        ))}
      </ImageList>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('masonry prop', () => {
    const { container } = render(<ImageList masonry />);
    expect(container.firstChild).toHaveClass('mdc-image-list--masonry');
  });

  test('masonry withTextProtection', () => {
    const { container } = render(<ImageList withTextProtection />);
    expect(container.firstChild).toHaveClass(
      'mdc-image-list--with-text-protection'
    );
  });
});
