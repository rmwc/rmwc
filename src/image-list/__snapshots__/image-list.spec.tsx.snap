// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList renders 1`] = `
<DocumentFragment>
  <ul
    class="mdc-image-list"
  >
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/1.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/2x3/1.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/4.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/5.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/6.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/2x3/2.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/8.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/11.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/2x3/5.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/13.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/14.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/2x3/6.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/15.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/3x2/16.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
    <li
      class="mdc-image-list__item"
      style="margin: 2px; width: calc(100% / 5 - 4.2px);"
    >
      <div
        class="mdc-image-list__image-aspect-container"
      >
        <img
          class="mdc-image-list__image"
          src="https://material-components-web.appspot.com/images/photos/2x3/7.jpg"
        />
      </div>
      <div
        class="mdc-image-list__supporting"
      >
        <div
          class="mdc-image-list__label"
        >
          Text label
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
